lib/ansible/plugins/action/gather_facts.py

```diff
@@ -68,7 +68,7 @@ def run(self, tmp=None, task_vars=None):
         if 'smart' in modules:
             connection_map = C.config.get_config_value('CONNECTION_FACTS_MODULES', variables=task_vars)
             network_os = self._task.args.get('network_os', task_vars.get('ansible_network_os', task_vars.get('ansible_facts', {}).get('network_os')))
-            modules.extend([connection_map.get(network_os or self._connection._load_name, 'setup')])
+            modules.extend([connection_map.get(network_os or self._connection._load_name, 'ansible.legacy.setup')])
```

Comment:
Another on in this file:

```
107:                    res = self._execute_module(module_name='async_status', module_args=poll_args, task_vars=task_vars, wrap_async=False)
```

Reply:
Do we need to do some additional checking on the values we get from here?:
```python
66:     modules = C.config.get_config_value('FACTS_MODULES', variables=task_vars)
```

In theory, that should only be _extra_ facts modules, but just wanted to ask. What if somebody puts `setup` or `gather_facts` in that list?

Reply:
The trick is that since that's an arbitrary user-supplied list, we wouldn't necessarily know which collection to prefix with... I suppose I could add a warning for unqualified values, but beyond that, I'm not sure what else we could do.